clang/lib/CodeGen/CGCall.cpp

```diff
@@ -1836,6 +1837,45 @@ static bool HasStrictReturn(const CodeGenModule &Module, QualType RetTy,
          Module.getLangOpts().Sanitize.has(SanitizerKind::Return);
 }
 
+llvm::fp::FPAccuracy convertFPAccuracy(StringRef FPAccuracy) {
+  StringRef AccuracyVal;
+  if (FPAccuracy == "high")
+    return llvm::fp::FPAccuracy::High;
+  else if (FPAccuracy == "medium")
+    return llvm::fp::FPAccuracy::Medium;
+  else if (FPAccuracy == "low")
+    return llvm::fp::FPAccuracy::Low;
+  else if (FPAccuracy == "sycl")
+    return llvm::fp::FPAccuracy::SYCL;
+  else if (FPAccuracy == "cuda")
+    return llvm::fp::FPAccuracy::CUDA;
+  else
+    llvm_unreachable("Unexpected type for FPAccuracy");
+}
+
+void CodeGenModule::getDefaultFunctionFPAccuracyAttributes(
+    StringRef Name, llvm::AttrBuilder &FuncAttrs, unsigned ID,
+    const llvm::Type *FuncType) {
+  for (const auto &M : getLangOpts().FPAccuracyMap) {
+    llvm::StringSet<> FuncOwnAttrs;
+    StringRef FPAccuracyVal = llvm::fp::getAccuracyForFPBuiltin(
+        ID, FuncType, convertFPAccuracy(M.second));
+    FuncAttrs.addAttribute("fpbuiltin-max-error=", FPAccuracyVal);
```

Comment:
You should check here for getAccuracyForFPBuiltin returning an empty StringRef. Also on line 1873 below.

Reply:
Okay. Using the wrong value of fp-accuracy is diagnosed early.  I can't find a case where we can arrive here with a null string.